packages/ckeditor5-utils/src/dom/rect.js

```diff
@@ -406,3 +406,25 @@ function isBody( elementOrRange ) {
 
 	return elementOrRange === elementOrRange.ownerDocument.body;
 }
+
+// Copies size properties from the `source` to the `rect`.
+//
+// @private
+// @param {module:utils/dom/rect~Rect} rect Target rect.
+// @param {Range} source
+function copyRangeRectProperties( rect, source ) {
+	const rangeRects = Rect.getDomRangeRects( source );
+	const combinedRect = rangeRects[ 0 ];
+
+	for ( let i = 1; i < rangeRects.length; i++ ) {
+		combinedRect.right = Math.max( combinedRect.right, rangeRects[ i ].right );
+		combinedRect.left = Math.min( combinedRect.left, rangeRects[ i ].left );
+		combinedRect.bottom = Math.max( combinedRect.bottom, rangeRects[ i ].bottom );
+		combinedRect.top = Math.min( combinedRect.top, rangeRects[ i ].top );
+	}
+
+	combinedRect.width = combinedRect.right - combinedRect.left;
+	combinedRect.height = combinedRect.bottom - combinedRect.top;
+
+	copyRectProperties( rect, combinedRect );
+}
```

Comment:
In the table package we have a helper function that is very similar (the same purpose but different approach). Maybe we could unify those to one helper function? 

https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-table/src/utils/ui/contextualballoon.js#L123-L144

Reply:
Good idea, added #7858 that will also be closed by this PR.

I had a small problem with the naming, first wanted to make it a `Range#expand()` member, but the name didn't feel right. Ended up with static `Range.getBoundingRect()` which feels just right :ok\_hand:

Also I simplified mentioned contextualballoon logic a little bit as the way how `createBoundingRect` [internal logic was implemented in the `getBalloonCellPositionData` function](https://github.com/ckeditor/ckeditor5/blob/1738d0f2296cfc9699cb1d78d7c459f60f1465e0/packages/ckeditor5-table/src/utils/ui/contextualballoon.js#L91-L93) was a little confusing.

Reply:
I guess you mean `Rect` not `Range` 😉